lib/p2p/Pool.ts

```diff
@@ -333,12 +333,36 @@ class Pool extends EventEmitter {
     this.peers.forEach(peer => peer.close());
   }
 
-  private listen = (port: number): void => {
-    this.server.listen(port, '0.0.0.0');
+  /**
+   * Start listening for incoming p2p connections on the given port.
+   * @return a promise that resolves once the server is listening, or rejects if it fails to listen
+   */
+  private listen = (port: number) => {
```

Comment:
Shouldn't we define the return type even if it is just `Promise<void>`?

Reply:
It's implied in this case, I've been letting return types be implicitly defined for short/simple methods. Since the first line starts with `return new Promise<void>` I think it's clear enough in this case, but let me know if you think differently.